components/callbacks/executors_test.go

```diff
@@ -223,6 +237,168 @@ func TestProcessInvocationTask_Outcomes(t *testing.T) {
 	}
 }
 
+func TestProcessInvocationTaskHsm_Outcomes(t *testing.T) {
+	cases := []struct {
+		name                  string
+		expectedError         error
+		destinationDown       bool
```

Comment:
We won't use the circuit breaker for these tasks, remove this field.

Reply:
We may want to consider failing callback invocation tasks though instead of storing failed results to leverage ephemeral timers instead of just persistent ones but let's keep the persistent timer for now and consider optimizing later.
If we do decide to add a circuit breaker at some later point, it would work by marking the state machine's shard as the destination that's down.